gcloud/error_reporting/client.py

```diff
@@ -0,0 +1,109 @@
+    a Logging client.
+
+    :type project: string
+    :param project: the project which the client acts on behalf of. If not
+                    passed falls back to the default inferred from the
+                    environment.
+
+    :type credentials: :class:`oauth2client.client.OAuth2Credentials` or
+                       :class:`NoneType`
+    :param credentials: The OAuth2 Credentials to use for the connection
+                        owned by this client. If not passed (and if no ``http``
+                        object is passed), falls back to the default inferred
+                        from the environment.
+
+    :type http: :class:`httplib2.Http` or class that defines ``request()``.
+    :param http: An optional HTTP object to make requests. If not passed, an
+                 ``http`` object is created that is bound to the
+                 ``credentials`` for the current object.
+
+    :type service: str
+    :param service: An identifier of the service, such as the name of the
+                    executable, job, or Google App Engine module name. This
+                    field is expected to have a low number of values that are
+                    relatively stable over time, as opposed to version,
+                    which can be changed whenever new code is deployed.
+
+    :raises: :class:`ValueError` if the project is neither passed in nor
+             set in the environment.
+    """
+
+    def __init__(self, project=None,
+                 credentials=None,
+                 http=None,
+                 service=None):
+        self.logging_client = gcloud.logging.client.Client(
+            project, credentials, http)
+        self.service = service
+
+    DEFAULT_SERVICE = 'python'
+
+    def _get_default_service(self):
+        """Returns the service to use on method calls that don't specify an
+        override.
+
+          :rtype: string
+          :returns: The default service for error reporting calls
+        """
+        if self.service:
+            return self.service
+        else:
+            return self.DEFAULT_SERVICE
+
+    def report_error(self, message="", service=None):
+        """ Reports the details of the latest exceptions to Stackdriver Error
+            Reporting.
+
+         https://cloud.google.com/error-reporting/docs/formatting-error-messages
+
+           :type message: str
+           :param message: An optional message to include with the exception
+                           detail
+
+           :type service: str
+           :param service: An identifier of the service, such as the name of
+                           the executable, job, or Google App Engine module
+                           name. This field is expected to have a low number
+                           of values that are relatively stable over time,
+                           as opposed to version, which can be changed
+                           whenever new code is deployed.
+
```

Comment:
https://cloud.google.com/error-reporting/docs/grouping

Yes we should have a call that takes most of the parameters in context. I took a shortcut as a baby step thinking we'd merge into feature branches but I can write that call now if you think I should. But basic idea is that you'd rarely actually use that lower-level call, rather we add some helper functions in here that fill it in for you. Not clear if you want things like code that pulls out the Django context and fills it in this library, but in my head we would put that into some contrib package in here.